### PR TITLE
improve watch API with useFieldArray delete

### LINKED ISSUE #2031: [v6] - watch returns defaultValue after moving an item in a fieldArray
**Describe the bug**
Not sure if this is an error or is it by design. Having a `fieldArray`, when you move one of the items,  the watch for the FA first returns the default value of the array and then the updated value. I'm not sure why the watch runs twice, but if it does, shouldn't it return the latest value instead of the default?

**To Reproduce**
1. add 4 items to the list
2. move 1 item
3. see console log (it shows first the watch result with the initial value of the FA and then the updated value)

**Codesandbox link (Required)**
https://codesandbox.io/s/react-hook-form-usefieldarray-mnemu?file=/src/index.js

### LINKED ISSUE #2052: Registering new field when form is invalid can make the form valid
**Describe the bug**
when the form is invalid and a new field is registered, the validation resolver is passed initial state, which may result in the form become incorrectly valid
i might be missing something since this library is very well thought out and i see the code that intentionally passes initial data when the form is invalid so please excuse me if i made a mistake

**To Reproduce**
1. Go to https://codesandbox.io/s/react-hook-form-validationresolver-g7ett?file=/src/App.js
2. Click "Include username" checkbox to cause form to be in invalid state
3. Click "Toggle username field" to show username
4. See error - form is is valid state/validation is passed initial form data, causing the form to be incorrectly valid

**Codesandbox link (Required)**
https://codesandbox.io/s/react-hook-form-validationresolver-g7ett?file=/src/App.js

**Expected behavior**
when registering new inputs when form is invalid, the form will be passed the current state(merged with initial state?)

**Desktop (please complete the following information):**
 - OS: macOS
 - Browser: chrome
 - Version 83

### PR DESCRIPTION
**Pull request by bluebill1049**
React Hook Form's form states live inside each input, which leads to difficultly to track `useFieldArray` inputs value during deletion, so this PR is aimed to improve this behavior. After this PR, user `fields` as `watch`'s defaultValue to assure `watch` return correct inputs value.

**usage:** 

```
const { fields } = useFieldArray()
watch('test', fields);
```

- remove additional function
- using `getValues()` in `useFieldArray` and `useForm` with `resolver`
- update unit tests

close #2031
close #2052

### CODE REVIEW

File: src/useForm.ts [hunk 1]
```diff
@@ -309,7 +305,7 @@ export function useForm<
         | InternalFieldName<TFieldValues>[],
     ) => {
       const { errors } = await resolverRef.current!(
-        getFieldArrayValueByName(fieldsRef.current),
+        getValues() as any,
```
Comment (bluebill1049) [inline src/useForm.ts:308]: @kotarella1110 can you please assist with this type issue?

**Review by JeromeDeLeon — DISMISSED** [review]
LGTM! I believe this doesn't affect how useWatch works for useFieldArray, right?

**Comment by bluebill1049** [comment]
> LGTM! I believe this doesn't affect how useWatch works for useFieldArray, right?

no, it doesn't. thanks for the review ❤️

**Comment by bluebill1049** [comment]
@keiya01 I love your 👀 icon LOL.

**Comment by keiya01** [comment]
👀  is very useful lol
I reviewed the PR! I think we should write test about this behavior.
Could I write test about this behavior?

**Comment by bluebill1049** [comment]
> 👀 is very useful lol
> I reviewed the PR! I think we should write test about this behavior.
> Could I write test about this behavior?

Bring your 🔥 🚀

File: src/useFieldArray.test.tsx [hunk 2]
```diff
@@ -11,6 +11,10 @@ jest.mock('./logic/generateId', () => ({
 }));
 
 describe('useFieldArray', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
```
Comment (keiya01) [inline src/useFieldArray.test.tsx:17]: sorry, I have one question.
Why did you add this code ?

Comment (bluebill1049) [reply src/useFieldArray.test.tsx:17]: One of the test pass when i run individual and failed when i run them all, i guess it was to do with mock function, so i just reset it.

Comment (keiya01) [reply src/useFieldArray.test.tsx:17]: I see. Thank you.

**Comment by keiya01** [comment]
@bluebill1049 
I'm sorry. Testing this behavior is too difficult. Therefore I will try to write test if I come up with any solution.🙏

**Review by keiya01 — APPROVED** [review]

**Comment by bluebill1049** [comment]
> @bluebill1049
> I'm sorry. Testing this behavior is too difficult. Therefore I will try to write test if I come up with any solution.🙏

no worries at all, thanks for trying out! we have plenty of time anyhow.
